ctru-rs/src/error.rs

```diff
@@ -17,12 +19,22 @@ impl From<i32> for Error {
 impl fmt::Debug for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::Os(err) => write!(f, "libctru result code: {:08X}", err),
+            Error::Os(err) => {
+                write!(f, "libctru result code: {:08X}:", err)?;
```

Comment:
This should either have a space at the end or be incorporated into the `debug_struct` call as a separate field (ex. `raw`).

Reply:
`debug_struct` adds a space by default since it normally is called with a name, e.g. `debug_struct("Foo")` becomes `Foo { .. }`, but I like the idea of a `raw` field... then maybe we just have something like
```rs
        match *self {
            Error::Os(err) => {
                f.debug_struct("Error")
                    .field("raw", &format!("{:#08X}", err))
                    .field("description", &R_DESCRIPTION(err))
                    .field("module", &R_MODULE(err))
                    .field("summary", &R_SUMMARY(err))
                    .field("level", &R_LEVEL(err))
                    .finish()
            }
        }
```

The downside of stuffing things into `field()` is they use their own default `Debug` impl, so there's not much customizing the format unless you print a string – and then, you'll get quotes around the value. But not a huge deal tbh

Reply:
I did find a bit of a hacky workaround, but it's not very customizable and not very nice... https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6ff1560fc0a8b9e4cf283856ed187c71

Reply:
By adding a space I meant after the colon before, because I think it would look like `libctru result code: -1:Error { ... }`

This also works, and is a little nicer:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ec0a9b23ebe2955318219845feac19d7
I discovered this after looking at this example: https://doc.rust-lang.org/stable/std/fmt/struct.Formatter.html#method.debug_struct